### PR TITLE
backstage: dont run percy unless correct label is applied on the PR

### PR DESCRIPTION
**Pull request by akomiqaia**
Creates PR template (which is very basic and need more love)
And introduces a git label to run workflows. If the git label is not `Run Percy` Percy jobs should not run.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## Describe your changes
```
Comment (frshwtr) [inline .github/pull_request_template.md:1]: Nice!

**Review by frshwtr — APPROVED** [review]

**Review by notlee — DISMISSED** [review]
I don't think these changes meet this part of the ticket?

>Maintain existing behaviour for chore and main Percy runs. This will ensure tests are always run before a release and screenshots are always up to date.

If we don't always run on merge to main, then the baseline screenshots will become out of date, and future PR runs may compare against old screenshots.

**Comment by frshwtr** [comment]
> I don't think these changes meet this part of the ticket?
> 
> > Maintain existing behaviour for chore and main Percy runs. This will ensure tests are always run before a release and screenshots are always up to date.
> 
> If we don't always run on merge to main, then the baseline screenshots will become out of date, and future PR runs may compare against old screenshots.

@notlee If I'm reading correctly, the workflows will run on main, but will not execute the percy scripts due to the `if` statements? And your suggestion is that percy should still run on main every time?

**Comment by ka-tsu-ra** [comment]
Looks like you've added a condition to run the percy tasks if main branch is building. (I've not used github actions before so I'm trusting that you know `github.ref` and `refs/heads/main` work)

But should the condition be using a `==` i.e. `github.ref == 'refs/heads/main'` rather than `github.ref = 'refs/heads/main'`?

**Comment by frshwtr** [comment]
> Looks like you've added a condition to run the percy tasks if main branch is building. (I've not used github actions before so I'm trusting that you know `github.ref` and `refs/heads/main` work)
> 
> But should the condition be using a `==` i.e. `github.ref == 'refs/heads/main'` rather than `github.ref = 'refs/heads/main'`?

Yes you're right. Good spot, thanks :)

**Comment by akomiqaia** [comment]
This would trigger anytime we merge with the main right? And this might cause us to rerun Percy jobs even when we don't have a tag on it right? My assumption was that since we merge PR with Percy label then if statement should be true and Percy screenshots should get updated. But if we don't apply Percy label that means we don't have to run Percy jobs.

**Comment by frshwtr** [comment]
> This would trigger anytime we merge with the main right? And this might cause us to rerun Percy jobs even when we don't have a tag on it right? My assumption was that since we merge PR with Percy label then if statement should be true and Percy screenshots should get updated. But if we don't apply Percy label that means we don't have to run Percy jobs.

`ref` on this context object references the branch which triggered the Github Action. We would want this behaviour as we still need to run on main once merged so that screenshots are up to date. See [here](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context)

**Comment by ka-tsu-ra** [comment]
From my non-origami pov this looks ok :)

**Review by ka-tsu-ra — APPROVED** [review]

**Pull request merged**